beluga/include/beluga/motion/ackerman_drive_model.hpp

```diff
@@ -0,0 +1,269 @@
+      const {
+    // Calculate velocities from poses
+    const auto [linear_velocity, angular_velocity] = calculate_velocities(pose, previous_pose, delta_time);
+
+    using DistributionParam = typename std::normal_distribution<double>::param_type;
+
+    // Velocity noise parameters (following velocity motion model from Probabilistic Robotics)
+    const auto linear_velocity_params = DistributionParam{
+        linear_velocity, std::sqrt(
+                             params_.translation_noise_from_translation * std::abs(linear_velocity) +
+                             params_.translation_noise_from_rotation * std::abs(angular_velocity))};
+
+    const auto angular_velocity_params = DistributionParam{
+        angular_velocity, std::sqrt(
+                              params_.rotation_noise_from_translation * std::abs(linear_velocity) +
+                              params_.rotation_noise_from_rotation * std::abs(angular_velocity))};
+
+    // Additional orientation noise (gamma_hat) using rotation parameters
+    const auto gamma_params = DistributionParam{
+        0.0,  // zero mean
+        std::sqrt(
+            params_.orientation_noise_from_translation * std::abs(linear_velocity) +
+            params_.orientation_noise_from_rotation * std::abs(angular_velocity))};
+
+    return [=](const auto& state, auto& gen) {
+      static thread_local auto distribution = std::normal_distribution<double>{};
+
+      // Sample noisy velocities
+      const auto v_hat = distribution(gen, linear_velocity_params);
+      const auto omega_hat = distribution(gen, angular_velocity_params);
+      const auto gamma_hat = distribution(gen, gamma_params);
+
+      // Apply velocity motion model
+      return apply_velocity_motion(state, v_hat, omega_hat, gamma_hat, delta_time);
+    };
+  }
+
+  /// Calculate linear and angular velocities from two poses and delta time
+  std::pair<double, double>
+  calculate_velocities(const Sophus::SE2d& pose, const Sophus::SE2d& previous_pose, double delta_time) const {
+    // Euclidean distance (chord length between poses)
+    const auto translation = pose.translation() - previous_pose.translation();
+    const double chord_distance = translation.norm();
+
+    // Angular velocity from orientation change
+    const auto angular_change = pose.so2() * previous_pose.so2().inverse();
+    const double angle_change = angular_change.log();
+    const double angular_velocity = angle_change / delta_time;
+
+    // Determine direction sign (forward/backward motion)
+    const auto forward_direction =
+        Eigen::Vector2d{std::cos(previous_pose.so2().log()), std::sin(previous_pose.so2().log())};
+    const double dot_product = translation.dot(forward_direction);
+    const double sign = (dot_product >= 0) ? 1.0 : -1.0;
+
+    // Linear velocity calculation
+    double linear_velocity = 0.0;
+
+    if (std::abs(angle_change) > 1e-6) {
```

Comment:
@fbattocchia nit: consider using an proper epsilon for this ie. `std::numeric_limits<double>::epsilon()`. Same elsewhere.